### PR TITLE
Searchbar

### PR DESCRIPTION
**Pull request by limpa105**
# Pull Request Process
1. Ensure that your branch is as up-to-date with main as possible 
2. Fill out this template
3. Have at least 2 other people to get PR approval comments (could just add everybody on team)
4. Create PR

# Changes
## What changes did you make?
Search bar working! Also routed the buttons on home page and the id is now embedded in the link when you click on a recipe  

# Verification
## How did you ensure that your changes work?
Tested locally 

# Screenshots
## Please post screenshots of your changes working if possibl
Screen Shot 2021-11-17 at 5 22 00 PM
e

### CODE REVIEW

File: source/search_page/assets/searchRecipe.js [hunk 1]
```diff
@@ -79,21 +92,21 @@ class SearchRecipe extends HTMLElement {
         const timeNumb = document.createElement('p')
         timeNumb.setAttribute('class', 'recipe-time');
 
-        timeNumb.innerHTML = "14";
+        timeNumb.innerHTML = data.readyInMinutes + "\n min";
 
 
-        recipeImg.setAttribute('src','images/pasta.jpeg');
+        recipeImg.setAttribute('src',data.image);
```
Comment (nlpham) [inline source/search_page/assets/searchRecipe.js:98]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8435053)
```suggestion
        recipeImg.setAttribute("src",data.image);
```

File: source/search_page/assets/searchRecipe.js [hunk 2]
```diff
@@ -52,25 +54,36 @@ class SearchRecipe extends HTMLElement {
...
-        recipe.setAttribute('href',"");
 
         const recipeTitle = document.createElement('p');
-        recipeTitle.setAttribute('href',"");
+        //recipeTitle.setAttribute('href', `../recipe-individual/inex.html?${data.id}`);
         recipeTitle.setAttribute('class','headline');
 
+        const link = document.createElement('a')
+        link.setAttribute('href', `../recipe-individual/index.html?${data.id}` )
+        link.setAttribute('class', 'link');
```
Comment (nlpham) [inline source/search_page/assets/searchRecipe.js:84]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8435053)
```suggestion
        link.setAttribute("class", 'link');
```

File: source/search_page/assets/searchRecipe.js [hunk 3]
```diff
@@ -52,25 +54,36 @@ class SearchRecipe extends HTMLElement {
...
         const recipe = document.createElement('article');
-        recipe.setAttribute('href',"");
 
         const recipeTitle = document.createElement('p');
-        recipeTitle.setAttribute('href',"");
+        //recipeTitle.setAttribute('href', `../recipe-individual/inex.html?${data.id}`);
         recipeTitle.setAttribute('class','headline');
 
+        const link = document.createElement('a')
+        link.setAttribute('href', `../recipe-individual/index.html?${data.id}` )
```
Comment (nlpham) [inline source/search_page/assets/searchRecipe.js:83]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8435053)
```suggestion
        link.setAttribute("href", `../recipe-individual/index.html?${data.id}` )
```

File: source/search_page/assets/searchRecipe.js [hunk 4]
```diff
@@ -52,25 +54,36 @@ class SearchRecipe extends HTMLElement {
...
 
         const recipe = document.createElement('article');
-        recipe.setAttribute('href',"");
 
         const recipeTitle = document.createElement('p');
-        recipeTitle.setAttribute('href',"");
+        //recipeTitle.setAttribute('href', `../recipe-individual/inex.html?${data.id}`);
         recipeTitle.setAttribute('class','headline');
 
+        const link = document.createElement('a')
```
Comment (nlpham) [inline source/search_page/assets/searchRecipe.js:82]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8435053)
```suggestion
        const link = document.createElement("a")
```

File: source/search_page/assets/script.js [hunk 5]
```diff
@@ -23,4 +21,32 @@ function createRecipeCards() {
     
   
 }
+*/
+
+async function searchFetchRecipes(searchBar) {
+  let searchString = `https://api.spoonacular.com/recipes/complexSearch?apiKey=03722052291e4f84bce1021acd82624f&${searchBar}&addRecipeInformation=True`;
+  console.log(searchString)
```
Comment (nlpham) [inline source/search_page/assets/script.js:28]: Codacy has a fix for the issue: [Missing semicolon.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8435053)
```suggestion
  console.log(searchString);  
```

**Review by nhieubrian — APPROVED** [review]

File: source/search_page/assets/searchRecipe.js [hunk 6]
```diff
@@ -52,25 +55,36 @@ class SearchRecipe extends HTMLElement {
...
```
Comment (nlpham) [inline source/search_page/assets/searchRecipe.js:83]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8435053)
```suggestion
        const link = document.createElement("a")
```

File: source/search_page/assets/script.js [hunk 7]
```diff
@@ -23,4 +21,29 @@ function createRecipeCards() {
...
+    return;
+  }
+  let querystring = window.location.href.substring(questionLocation + 1);
+  let resultArray = await searchFetchRecipes(querystring).then((response) => {
+    return response.results;
+  });
+  for(let i=0; i< resultArray.length; i++){
+    let ele = document.createElement('search-recipe');
+    ele.data = resultArray[i];
+    document.querySelector('main').append(ele);
```
Comment (nlpham) [inline source/search_page/assets/script.js:45]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8435053)
```suggestion
    document.querySelector("main").append(ele);
```

File: source/search_page/assets/script.js [hunk 8]
```diff
@@ -23,4 +21,29 @@ function createRecipeCards() {
...
+
+  };
```
Comment (nlpham) [inline source/search_page/assets/script.js:47]: Codacy has a fix for the issue: [Unnecessary semicolon.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8435053)
```suggestion
  }
```

**Review by georgetanUCSD — APPROVED** [review]

**Pull request merged**